src/nbi/engine.py

```diff
@@ -703,79 +672,70 @@ def get_network(self):
 
         Parameters
         ----------
-        epoch : int
-            Epoch number to load state from.
+        state_dict : str or state dict
+            State dict or path to saved state dict containing three keys:
+            network_state_dict, x_scale, y_scale
 
         Returns
         -------
 
         """
-        path_round = os.path.join(self.directory, str(self.round))
-        path = os.path.join(path_round, str(epoch) + ".pth")
-        self.get_network().load_state_dict(torch.load(path, map_location=self.device))
+        if type(state_dict) == str:
+            state_dict = torch.load(state_dict, map_location=self.device)
+        model_state_dict = state_dict["model_state_dict"]
 
-    def set_params(self, network, x_scale, y_scale):
-        """
-        Load engine parameters from disk, including network weights and data pre-processing scales.
+        # Move x_scale and y_scale to CPU before converting to numpy arrays
+        x_scale = state_dict["x_scale"].cpu().numpy()
+        y_scale = state_dict["y_scale"].cpu().numpy()
 
-        Parameters
-        ----------
-        network : str
-            Path of network state dict.
-        x_scale : str or ndarray
-            Path of x-scale or x-scale array.
-        y_scale : str or ndarray
-            Path of y-scale or y-scale array.
+        self.x_mean = x_scale[0]
+        self.x_std = x_scale[1]
+        self.y_mean = y_scale[0]
+        self.y_std = y_scale[1]
+        self.get_network().load_state_dict(model_state_dict)
+
+    def get_params(self):
+        """
+        Saves the network weights and pre-processing scales to disk
 
         Returns
         -------
 
         """
-        if type(x_scale) == str:
-            x_scale = np.load(x_scale)
-        if type(y_scale) == str:
-            y_scale = np.load(y_scale)
+        x_scale = np.array([self.x_mean, self.x_std])
+        y_scale = np.array([self.y_mean, self.y_std])
 
-        self.x_mean = x_scale[0]
-        self.x_std = x_scale[1]
-        self.y_mean = y_scale[0]
-        self.y_std = y_scale[1]
-        self.get_network().load_state_dict(
-            torch.load(network, map_location=self.device)
-        )
+        # Convert numpy arrays to PyTorch tensors
+        x_scale_tensor = torch.from_numpy(x_scale)
+        y_scale_tensor = torch.from_numpy(y_scale)
+
+        # Assuming 'network' is your model
+        model_state_dict = copy.deepcopy(self.get_network().state_dict())
+
+        # Create a new dictionary to store model state and additional tensors
+        state_dict = {
+            "model_state_dict": model_state_dict,
+            "x_scale": x_scale_tensor,
+            "y_scale": y_scale_tensor,
+        }
+        return state_dict
 
-    def save_state_dict(self):
+    def save_params(self, path):
         """
         Saves the network weights and pre-processing scales to disk
 
         Returns
         -------
 
```

Comment:
```suggestion
```

Reply:
Are you suggesting that all functions without returns should not have "Returns" in the doctoring?